pulsar-client-cpp/lib/HTTPLookupService.cc

```diff
@@ -0,0 +1,206 @@
+            curl_easy_cleanup(handle);
+            return;
+        }
+        struct curl_slist *list = NULL;
+        if (authDataContent->hasDataForHttp()) {
+            list = curl_slist_append(list, authDataContent->getHttpHeaders().c_str());
+        }
+        curl_easy_setopt(handle, CURLOPT_HTTPHEADER, list);
+
+        // Make get call to server
+        res = curl_easy_perform(handle);
+
+        // Free header list
+        curl_slist_free_all(list);
+
+        switch(res) {
+            case CURLE_OK:
+                LOG_DEBUG("Response received successfully for url " << completeUrl);
+                promise.setValue((requestType == PartitionMetaData) ? parsePartitionData(responseData) : parseLookupData(responseData));
+                break;
+            case CURLE_COULDNT_CONNECT:
+            case CURLE_COULDNT_RESOLVE_PROXY:
+            case CURLE_COULDNT_RESOLVE_HOST:
+            case CURLE_HTTP_RETURNED_ERROR:
+                LOG_ERROR("Response failed for url "<<completeUrl << ". Error Code "<<res);
+                promise.setFailed(ResultConnectError);
+                break;
+            case CURLE_READ_ERROR:
+                LOG_ERROR("Response failed for url "<<completeUrl << ". Error Code "<<res);
+                promise.setFailed(ResultReadError);
+                break;
+            case CURLE_OPERATION_TIMEDOUT:
+                LOG_ERROR("Response failed for url "<<completeUrl << ". Error Code "<<res);
+                promise.setFailed(ResultTimeout);
+                break;
+            default:
+                LOG_ERROR("Response failed for url "<<completeUrl << ". Error Code "<<res);
+                promise.setFailed(ResultLookupError);
+                break;
+        }
+        curl_easy_cleanup(handle);
+    }
+
+    LookupDataResultPtr HTTPLookupService::parsePartitionData(const std::string &json) {
+        Json::Value root;
+        Json::Reader reader;
+        if (!reader.parse(json, root, false)) {
+            LOG_ERROR("Failed to parse json of Partition Metadata: " << reader.getFormatedErrorMessages()
+                    << "\nInput Json = " << json);
+            return LookupDataResultPtr();
+        }
+        LookupDataResultPtr lookupDataResultPtr = boost::make_shared<LookupDataResult>();
+        lookupDataResultPtr->setPartitions(root.get("partitions", 0).asInt());
+        return lookupDataResultPtr;
+    }
+
+    LookupDataResultPtr HTTPLookupService::parseLookupData(const std::string &json) {
+        Json::Value root;
+        Json::Reader reader;
+        if (!reader.parse(json, root, false)) {
+            LOG_ERROR("Failed to parse json : " << reader.getFormatedErrorMessages()
+                    << "\nInput Json = " << json);
+            return LookupDataResultPtr();
+        }
+        const std::string defaultNotFoundString = "Url Not found";
+        const std::string brokerUrl = root.get("brokerUrl", defaultNotFoundString).asString();
```

Comment:
Used get instead of find since the json_cpp version in yahoo env doesn't support the find API.